Jenkinsfile

```diff
@@ -1,69 +1,81 @@
-#!groovy
-
 @Library('pipeline-library') _
 
 def img
 
 node {
-    stage('build') {
-        checkout(scm)
-        img = buildApp(name: 'hypothesis/lms')
-    }
-
-    stage('test') {
-        hostIp = sh(script: 'facter ipaddress_eth0', returnStdout: true).trim()
-
-        postgres = docker.image('postgres:9.4').run('-P -e POSTGRES_DB=lmstest')
-        databaseUrl = "postgresql://postgres@${hostIp}:${containerPort(postgres, 5432)}/lmstest"
-
-        try {
-            testApp(image: img, runArgs: "-u root -e TEST_DATABASE_URL=${databaseUrl} -e CODECOV_TOKEN=${credentials('LMS_CODECOV_TOKEN')}") {
-                sh 'apk add build-base postgresql-dev python3-dev yarn'
-                sh 'pip3 install -q tox>=3.8.0'
+    runArgs = "-u root -e SITE_PACKAGES=true"
 
-                // Hack to work around a race condition and make sure that
-                // .tox/.tox/bin/tox gets created before any of the parallelised
-                // `make` commands below try to access it.
-                sh 'cd /var/lib/lms && tox -e py36 --notest'
+    stage("Build") {
+        checkout scm
+        img = buildApp(name: "hypothesis/lms")
+    }
 
-                sh 'cd /var/lib/lms && make -j `nproc` checkformatting lint backend-tests'
-                sh 'cd /var/lib/lms && make coverage codecov'
+    parallel failFast: true,
+    "Formatting": {
+        stage("Formatting") {
```

Comment:
You have to say the stage's name (`"Formatting"`) twice. There will be some way to avoid this with more usage of the Groovy language but :man_shrugging: